src/Components/Server/src/Circuits/CircuitPrerenderer.cs

```diff
@@ -71,8 +76,14 @@ public async Task<ComponentPrerenderResult> PrerenderComponentAsync(ComponentPre
                 Prerendered = true
             });
 
+            var record = JsonSerializer.ToString(new PrerenderedComponentRecord(
+                    circuitHost.CircuitId.Replace("--", "__"), // We need to do this due to the fact that -- is not allowed within HTML comments and HTML doesn't encode '-'.
```

Comment:
I was pretty sure, I saw `_` in the generated circuitId, so I looked it up:
https://github.com/aspnet/AspNetCore/blob/f35564ba0635da588214791b3662085bd30d243e/src/Shared/WebEncoders/WebEncoders.cs#L385

Reply:
`CircuitIdFactory` -> `Microsoft.AspNetCore.WebUtilities.Base64UrlTextEncoder` -> `WebEncoders`